### PR TITLE
JAVA-2922: Switch to modern framing format inside a channel handler

### PR DESCRIPTION
**Pull request by adutra**

### CODE REVIEW

**Comment by adutra** [comment]
@beobal I can't assign you for review but could you please take a look? Thanks!

**Comment by beobal** [comment]
@adutra will do ASAP

**Review by beobal — APPROVED** [review]
LGTM

File: driver-core/src/main/java/com/datastax/driver/core/FramingFormatHandler.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  private final Connection.Factory factory;
+
+  FramingFormatHandler(Connection.Factory factory) {
+    this.factory = factory;
+  }
+
+  @Override
+  protected void decode(ChannelHandlerContext ctx, Frame frame, List<Object> out) throws Exception {
+    boolean handshakeSuccessful =
+        frame.header.opcode == Type.READY.opcode || frame.header.opcode == Type.AUTHENTICATE.opcode;
```
Comment (olim7t) [inline driver-core/src/main/java/com/datastax/driver/core/FramingFormatHandler.java:41]: The protocol allows `OPTIONS` as the first-ever message, so shouldn't we test for `SUPPORTED` here as well? Or maybe v5 does not switch to modern framing for that first `SUPPORTED` response? (I must confess I don't remember, it's been a while)

Alternatively, we could test for `opcode != ERROR` like in the original code.

**Review by olim7t — APPROVED** [review]
LGTM, that's an elegant way to address the issue.

File: driver-core/src/main/java/com/datastax/driver/core/FramingFormatHandler.java (hunk 1 above)
Comment (adutra) [reply driver-core/src/main/java/com/datastax/driver/core/FramingFormatHandler.java:41]: The specs say that only READY and AUTHENTICATE triggers trigger a framing format change:

> The initial STARTUP message and any OPTIONS messages which precede it are expected to be unframed. Likewise, the responses returned by the server, SUPPORTED in response to OPTIONS and either READY or AUTHENTICATE in response to STARTUP are transmitted unframed. After sending the READY or AUTHENTICATE response to a STARTUP message, the server will begin encoding and decoding all further transmissions according to the protocol version of that STARTUP message.

That's why I figured that detecting those 2 response types would make more sense than detecting an ERROR response.

Comment (olim7t) [reply driver-core/src/main/java/com/datastax/driver/core/FramingFormatHandler.java:41]: Right, we're good then 👍

**Pull request merged**